client/state/reader/conversations/reducer.js

```diff
@@ -42,8 +43,32 @@ export const items = createReducer(
 			return newState;
 		},
 		[ READER_CONVERSATION_UPDATE_FOLLOW_STATUS ]: ( state, action ) => {
+			const stateKey = key( action.payload.siteId, action.payload.postId );
+
+			// If followStatus is null, remove the key from the state map entirely
+			if ( action.payload.followStatus === null ) {
+				return omit( state, stateKey );
+			}
+
 			const newState = assign( {}, state, {
-				[ key( action.payload.siteId, action.payload.postId ) ]: action.payload.followStatus,
+				[ stateKey ]: action.payload.followStatus,
+			} );
+
+			return newState;
+		},
+		[ READER_POSTS_RECEIVE ]: ( state, action ) => {
+			if ( ! action.posts ) {
+				return state;
+			}
+
+			let newState = state;
+
+			forEach( action.posts, post => {
+				if ( post.is_following_conversation ) {
+					newState = assign( {}, newState, {
```

Comment:
i don't know if this is actually a perf issue or not, but having an `assign` within a for loop makes more work for the garbage collector than we need to.

thinking out loud, we could do something like:
```js

// accumulate changes into newState
const newState = {};
forEach( action.posts, post => {
  if ( post.is_... ) {
    newState[ key( ...) ] = CONVERSATION_FOLLOW_STATUS_FOLLOWING;
  }
} )

// if there are no changes then return old state
if ( newState.size() === 0 ) {
 return state;
}

// if we have changes to make then return new merged object
return { ...state, ...newState };
```

Reply:
Much better 👍